src/main/java/carpet/utils/TypedField.java

```diff
@@ -0,0 +1,99 @@
+package carpet.utils;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+import org.apache.commons.lang3.ClassUtils;
+
+/**
+ * <p>A wrapper around a {@link Field} that uses a {@link VarHandle} and {@link MethodHandle}s to retrieve and set the contents of the field</p>
+ * 
+ * <p>It keeps the memory semantics of {@code volatile} if the field had that modifier.</p>
+ *
+ * @param <T> The {@link Field}'s type
+ */
+public final class TypedField<T> {
```

Comment:
Not sure about the use of this class

Reply:
I originally wanted to use it for some other stuff, but ended up just for this, true that it's not that useful for just one thing, I'll revert it to a regular field.

The thing was that this class not only kept the `<T>` instead of returning/accepting an untyped `Object`, but also checked access and everything at creation, and given it used Method/VarHandle it should be a tiny bit faster (though it's not used in any performance-heavy area so it doesn't matter).